layers/+lang/haskell/README.org

```diff
@@ -464,52 +468,53 @@ snippet into your =dotspacemacs/user-config= function.
 The error list is only set by flycheck. You are probably seeing errors
 highlighted by either ghc-mode or haskell-mode. Check the [[#flycheck-doesnt-work][Flycheck doesn't work]]
 section.
 
 ** Flycheck doesn't work
-You can check what is wrong with flycheck with the =flycheck-compile= command.
-This will show you the exact command line used and its output.
+You can use the =flycheck-compile= command to check what's wrong with flycheck.
+This will show you the exact command line that's used, and its output.
 
-If you are using stack, check the [[#flycheck-doesnt-work-with-stack][Flycheck doesn't work with =stack=]] section.
+If you are using =stack=, check the [[#flycheck-doesnt-work-with-stack][Flycheck doesn't work with =stack=]] section.
 
 ** Flycheck doesn't work with =stack=
-First check that flycheck uses the correct checker and all the paths are
+First check that flycheck uses the correct checker, and all the paths are
 properly configured using =flycheck-verify-setup= (~SPC e v~). You can force the
-checker with =flycheck-select-checker= (~SPC e s~) to ensure it uses
-=haskell-stack-ghc=. If it still doesn't work, it could be one of the following
-problems:
+checker with =flycheck-select-checker= (~SPC e s~) to ensure that it uses
+=haskell-stack-ghc=. If it still doesn't work, then it could be one of the
+following problems:
 
-- The stack build directory is wrong
+- The =stack= build directory is wrong
 - The project root is not set properly
 
-*** The stack build directory is wrong
-The path to the build directory containing some generated files is normally
-under =.stack-work/install/<os>/Cabal-<version>/build=.
+*** The =stack= build directory is wrong
+The path to the build directory, which contains some generated files, is
+normally under =.stack-work/install/<os>/Cabal-<version>/build=.
 
-However, the version of the cabal library used by stack to generate the
-directory name is not the version of the cabal library installed by stack, but
-the version of cabal associated to the GHC version. This error can happen after
-upgrading cabal or cabal-install. To check if this is the problem, compare the
-path name of the build path used by flycheck using =flycheck-compile= and
-compare it to to the actual path in the =.stack-work= directory. If they are
-different you need to reinstall ghc using the command =stack setup
---upgrade-cabal=.
+However the version of the cabal library that's used by =stack= to generate the
+directory name, is not the version of the cabal library that's installed by
+=stack=. But it's the version of cabal that's associated to the GHC version. This
+error can happen after upgrading cabal or cabal-install. To check if this is the
+problem, compare the path name of the build path that's used by flycheck, using
+=flycheck-compile=, and compare it to the actual path, in the =.stack-work=
+directory. If they are different, then you'll need to reinstall ghc, using the
```

Comment:
That's a lot of commas, and most of them aren't necessary.  It would read better without the commas before the two instances of "using" and the comma after "actual path".

Reply:
It's probably better if you write the correct sentence, since I have a case of the "commas", and don't know which to delete or keep. 😄 

This is what I came up with:
This error can happen after upgrading cabal or cabal-install. To check
if this is the problem, compare the path name of the build path that's used by
flycheck using =flycheck-compile= and compare it to the actual path in the
=.stack-work= directory. If they are different then you'll need to reinstall ghc
using the command =stack setup --upgrade-cabal=.

Reply:
What you came up with looks good.  (I would keep the comma in "If they are different," but it's not necessarily needed.)